texar/modules/encoders/xlnet_encoders.py

```diff
@@ -0,0 +1,602 @@
+
+        "reuse_len": int
+            Length of the memory that can be re-used
+
+        "initializer" : dict, optional
+            Hyperparameters of the default initializer that initializes
+            variables created in this module.
+            See :func:`~texar.core.get_initializer` for details.
+
+        "num_heads": int
+            Number of heads in the attention
+
+        "hidden_dim": int
+            Hidden dimension of the embeddings
+
+        "head_dim": int
+            Size of the vectors after head projection.
+
+        "dropout": float
+            Dropout rate for layers
+
+        "attention_dropout": float
+            Dropout rate for attention layers
+
+        "use_segments": bool
+            Boolean to indicate if the input has segments
+
+        "ffn_inner_dim": int
+            Dimension of PositionWise FF network's hidden layer
+
+        "activation": str or callable
+            Activation function applied to the output of the PositionWise FF.
+            See :func:`~texar.core.get_activation_fn` for more details.
+
+        "vocab_size" : int
+            The vocabulary size of `inputs` in `XLNet`.
+
+        "max_seq_len": int
+            Maximum len of the sequence allowed in one segment
+
+        "name" : str
+            Name of the module.
+        """
+
+        return {
+            "name": "xlnet_encoder",
+            'pretrained_model_name': 'xlnet-base-cased',
+            "untie_r": True,
+            "num_layers": 12,
+            "mem_len": 0,
+            "reuse_len": 0,
+            # initializer
+            "initializer": None,
+            # layer
+            "num_heads": 12,
+            "hidden_dim": 768,
+            "head_dim": 64,
+            "dropout": 0.1,
+            "attention_dropout": 0.1,
+            "use_segments": True,
+            # ffn
+            "ffn_inner_dim": 3072,
+            "activation": 'gelu',
+            # embedding
+            "vocab_size": 32000,
+            "max_seq_len": 512,
+            '@no_typecheck': ['pretrained_model_name']
+        }
+
+    def param_groups(self, lr=None, lr_layer_scale=1.0,
+                     decay_base_params=False):
+        r"""Create parameter groups for optimizers. When
+        :attr:`lr_layer_decay_rate` is not 1.0, parameters from each layer form
+        separate groups with different base learning rates.
+
+        Args:
+            lr (float): The learning rate. Can be omitted if
+                :attr:`lr_layer_decay_rate` is 1.0.
+            lr_layer_scale (float): Per-layer LR scaling rate. The `i`-th layer
+                will be scaled by `lr_layer_scale ^ (num_layers - i - 1)`.
+            decay_base_params (bool): If `True`, treat non-layer parameters
+                (e.g. embeddings) as if they're in layer 0. If `False`, these
+                parameters are not scaled.
+
+        Returns:
+            The parameter groups, used as the first argument for optimizers.
```

Comment:
This statement is too vague. Pls explain more (what's the data structure? what's each field?) and give an example of how to use this APIs.

Reply:
If the doc of pytorch version is also this vague, pls update it as well.

Reply:
Fixed in fbcdc5d. Issue in texar-pytorch https://github.com/asyml/texar-pytorch/issues/145